api/filters/check_instance_visibility_filter.go

```diff
@@ -60,32 +60,47 @@ func (f *serviceInstanceVisibilityFilter) Run(req *web.Request, next web.Handler
 		return next.Handle(req)
 	}
 
+	criteria := []query.Criterion{
+		query.ByField(query.EqualsOrNilOperator, platformIDProperty, types.SMPlatform),
+		query.ByField(query.EqualsOperator, planIDProperty, planID),
+	}
+
+	list, err := f.repository.List(ctx, types.VisibilityType, criteria...)
+	if err != nil && err != util.ErrNotFoundInStorage {
+		return nil, util.HandleStorageError(err, types.VisibilityType.String())
+	}
+
+	visibilityError := &util.HTTPError{
+		ErrorType:   "NotFound",
+		Description: "could not find such service plan",
+		StatusCode:  http.StatusNotFound,
+	}
+	if list.Len() == 0 {
+		return nil, visibilityError
+	}
+
 	tenantID := query.RetrieveFromCriteria(f.tenantIdentifier, query.CriteriaForContext(ctx)...)
 	if tenantID == "" {
 		log.C(ctx).Info("Tenant identifier not found in request criteria. Proceeding with the next handler...")
 		return next.Handle(req)
 	}
 
-	criteria := []query.Criterion{
-		query.ByField(query.EqualsOperator, platformIDProperty, types.SMPlatform),
-		query.ByField(query.EqualsOperator, planIDProperty, planID),
-		query.ByLabel(query.InOperator, f.tenantIdentifier, tenantID),
-	}
-
-	_, err := f.repository.Get(ctx, types.VisibilityType, criteria...)
-	if err != nil {
-		if err == util.ErrNotFoundInStorage {
-			return nil, &util.HTTPError{
-				ErrorType:   "NotFound",
-				Description: "could not find such service plan",
-				StatusCode:  http.StatusNotFound,
+	visibilities := list.(*types.Visibilities).Visibilities
+	for _, v := range visibilities {
```

Comment:
Do we have to have the loop primitive? In reality there's either going to be 0 visibilities or 1 (public or one with platform_id=SM)

Can we just return visibility error if 0, and otherwise get the 0-th index visibility and directly work on it? Because it's somewhat confusing having this loop, at first I thought that there would indeed be lots of visibilities being traversed